kafkazk/brokers.go

```diff
@@ -325,6 +333,11 @@ func (b BrokerMap) Update(bl []int, bm BrokerMetaMap) (*BrokerStatus, <-chan str
 					New:         true,
 				}
 				bs.New++
+				if b[id].Locality == "" {
+					b[id].RackMissing = true
+					bs.RackMissing++
+					msgs <- fmt.Sprintf("Broker %d does not has a rack.id defined", id)
+				}
```

Comment:
Dupe info from L284-288

Reply:
On second read, I see why this is in here twice. Maybe what we can do to avoid duplicate code:

- Remove it from this loop (which isn't a bad place to check for missing rack metadata, but is focused on merging the broker sets
- Remove it on L284 as well (which focuses on checking whether brokers were found at all in zookeeper)
- Add a single pass at the end that checks if `bm` is > 0 (i.e. we have a `BrokerMetaMap` and care about broker metadata from zookeeper), and if so, loops `b`, checks if any brokers have missing rack.ids.